secondary-server-1/src/main/java/com/replicated_log/secondary_server_1/controller/rest/SecondaryRestController.java

```diff
@@ -0,0 +1,70 @@
+package com.replicated_log.secondary_server_1.controller.rest;
+
+import com.replicated_log.secondary_server_1.model.Ack;
+import com.replicated_log.secondary_server_1.model.AckStatusCode;
+import com.replicated_log.secondary_server_1.model.Item;
+import com.replicated_log.secondary_server_1.service.SecondaryClient;
+import com.replicated_log.secondary_server_1.service.ItemService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Set;
+
+@RestController
+@RequestMapping(SecondaryRestController.SECONDARY_1_URL)
+@CrossOrigin("*")
+public class SecondaryRestController {
+
+    private final Logger LOG = LogManager.getLogger(SecondaryRestController.class);
+
+    public static final String SECONDARY_1_URL = "/secondary1";
+    public final String SECONDARY_SERVER_NAME = "Secondary1";
+
+    @Value("${secondary.server.baseurl}")
+    private String SECONDARY_SERVER_BASE_URL;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private SecondaryClient secondaryClient;
+
+    @GetMapping("/items")
+    public ResponseEntity<Set<Item>> findAllItems() {
+        Set<Item> items = itemService.getItems();
+        return ResponseEntity.ok(items);
+    }
+
+    @GetMapping("/items-demo")
+    public ResponseEntity<Set<Item>> findAllItemsForDemo() {
```

Comment:
what is the difference between findAllItems vs findAllItems?

Reply:
findAllItemsForDemo() method shows messages from Secondary`s storage in 'total order' according to Project`s Requirements:
If secondary has received messages [msg1, msg2, msg4], it shouldn’t display the message ‘msg4’ until the ‘msg3’ will be received

findAllItems() method shows all excisting messages from Secondary`s storage

Reply:
need to discuss